### ISSUE TITLE
The weird curve

### ISSUE THREAD

**Issue by nguyenvannghiem0312** [PiotrNawrot/nanoT5#42]
Hello, thank you for providing the community with a great framework. I did some experiments with the T5-small model on the C4 Vietnamese dataset, and I would like to get some feedback from you:

1. Since the T5 vocabulary isn't suitable for Vietnamese, I took the configuration from google/t5-v1_1-small and modified the code to use a Vietnamese tokenizer and adjusted its vocabulary size. Is this approach correct?
```python
def get_tokenizer(args, config):
    tokenizer = AutoTokenizer.from_pretrained(
        args.model.tokenizer,
        use_fast=True
    )
    tokenizer.model_max_length = int(1e9)

    if args.model.tokenizer != args.model.name:
        config.vocab_size = tokenizer.vocab_size

    return tokenizer
```
  
2. When I followed the above method and pretrained the model, using AdamWScale and cosine LR, I made some adjustments to the LR and noticed something interesting afterward.
- Lr = 2e-2
![image](https://github.com/user-attachments/assets/f555c06b-c4d7-4303-bebf-908a526c8b7d)
![image](https://github.com/user-attachments/assets/8f109ec2-f5f6-4018-a0cf-05fc81de1518)

- Lr = 2e-3
![image](https://github.com/user-attachments/assets/17a06baa-92b7-4ab2-bc90-773d2ea1451b)
![image](https://github.com/user-attachments/assets/8a98b03f-a38d-4001-bc2d-ce02e841ef02)

- Lr = 2e-4
![image](https://github.com/user-attachments/assets/b5a8c17f-eaee-4927-83db-4bf42910503a)
![eval2e4](https://github.com/user-attachments/assets/ce64e9db-2bea-4408-a00c-f329569d6875)

I have a few comments: The LR of 2e-2 seems too large, causing the model not to converge. The LR of 2e-4 is too small; although the model appears to converge, the results are not very good. As for the LR of 2e-3, the model's performance is better, but there is a "jump" at 20k steps that makes me curious about the reason why. I suspect that using an LR of 5e-3 could improve results compared to 2e-3, but the curve would likely be similar.

**Comment by PiotrNawrot** [comment]
1. Yeah, I think so that's all you need.
2. It's a long not-understood behaviour mentioned in #25 - during the development of nanoT5 I copied the initialisation schemas from different repos until I managed to reproduce numbers reported in the paper. What we've concluded so far is that the initialisation schemas available online, also in nanoT5, might not be optimal and they might cause training instabilities, and they might require Adafactor LR scaling (which I've added to nanoT5) to work. I've done some preliminary experiments where I was playing with the initialisation and at some point I was able to achieve similar results with plain AdamW - however, due to other duties I had to stop there and haven't picked it up since. You can find all related info in the issue I mentioned.


TLDR - Imo it's amazing research mini task to understand what's going on here and one can definitely learn a lot of interesting insights by doing so. If you don't have time to do so I'd just tune the LR to the one that works best and have "the bump" you mention and just go along with it.
